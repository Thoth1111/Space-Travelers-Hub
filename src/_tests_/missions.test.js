import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Missions from '../routes/missions';

jest.mock('../redux/missions/missionsSlice', () => ({
  fetchMissions: jest.fn(),
}));

const mockStore = configureMockStore([thunk]);

describe('Missions component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      missions: [
        {
          missionId: '7',
          missionName: 'Seven',
          description: 'Seven is Divine',
          booked: true,
        },
        {
          missionId: '11',
          missionName: 'Eleven',
          description: 'rhymes with seven',
          booked: false,
        },
      ],
    });
  });

  it('renders mission data', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(screen.getByText('Seven')).toBeInTheDocument();
    expect(screen.getByText('Seven is Divine')).toBeInTheDocument();
    expect(screen.getByText('Active Member')).toBeInTheDocument();
    expect(screen.getByText('Eleven')).toBeInTheDocument();
    expect(screen.getByText('rhymes with seven')).toBeInTheDocument();
    expect(screen.getByText('NOT A MEMBER')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});