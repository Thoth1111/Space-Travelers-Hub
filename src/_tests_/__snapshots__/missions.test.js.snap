// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component matches snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Seven
      </td>
      <td>
        Seven is Divine
      </td>
      <td>
        <span
          class="active-status"
        >
          <span>
            Active Member
          </span>
        </span>
      </td>
      <td>
        <button
          class="leave-btn"
          type="button"
        >
          Leave Mission
        </button>
      </td>
    </tr>
    <tr>
      <td>
        Eleven
      </td>
      <td>
        rhymes with seven
      </td>
      <td>
        <span
          class="inactive-status"
        >
          <span>
            NOT A MEMBER
          </span>
        </span>
      </td>
      <td>
        <button
          class="join-btn"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
